app/channel.go

```diff
@@ -1749,6 +1749,18 @@ func (a *App) UpdateChannelLastViewedAt(channelIds []string, userId string) *mod
 	return nil
 }
 
+//MarkChanelAsUnreadFromPost will take a post and set the channel as unread from that one.
+func (a *App) MarkChannelAsUnreadFromPost(postID string, userID string) (*model.ChannelUnread, *model.AppError) {
+
+	post, err := a.GetSinglePost(postID)
+	if err != nil {
+		return nil, err
+	}
+
+	unreadMentions := 1 // TODO: calculate this value, setting it to one for now.
```

Comment:
Maybe just set this to 0 for now? This is going to be merged into master, so we don't want to confuse people by having the mention badge appear before mention counting is ready.

Reply:
sure, it was mostly for peeking how it affected everything, I wasn't thinking about shipping this PR without the others in the epic, but it might be possible